### ISSUE TITLE
How do I run the backend without desktop

### ISSUE THREAD

**Issue by Ahmedouu** [TryQuiet/quiet#1922]
in the docs it says there is an entryNode.ts but I can't find it.

thanks,

**Comment by leblowl** [comment]
Thanks for the issue @Ahmedouu. It looks like the docs are out of date and it doesn't seem like there is an easy way to run them separately anymore without changes to the frontend code since it appears to spawn the backend automatically: https://github.com/TryQuiet/quiet/blob/2de5ba6fb5cd925277922d95d65ef1f42b1a9589/packages/desktop/src/main/main.ts#L355-L373

I'd definitely like to fix this and the docs.

Here is an example of how I was running the backend independently recently:
```bash
cd packages/backend
npm run webpack && DEBUG=* node ../backend-bundle/bundle.cjs -d 42241 -a '/home/lucas/.config/Quietdev' -r '/home/lucas/src/tryquiet/quiet/packages/desktop/node_modules/electron/dist/resources' -p desktop
```

**Comment by Ahmedouu** [comment]
@leblowl  great!  meanwhile can you tell me what is in the .config/Quietdev ? 

Thanks,

**Comment by holmesworcester** [comment]
That's the same as .config/Quiet in the built version. It's the default file storage location that .AppImage uses. We use Quietdev so that you can run a dev version of Quiet alongside production Quiet without the two interfering. 

Inside there is the Tor configuration files, IPFS and OrbitDB data, and a few other things.

**Comment by Ahmedouu** [comment]
when I run the backend I don't see libp2p starting.

this is what I see when running desktop: 
` backend:Libp2pService Initializing libp2p for QmcxNcsNPi6RJaMNffVz6yDRqNFTdBYWvXDcD1ptTQDKAo, bootstrapping with 1 peers +1ms`

I am confused, libp2p is in the backend yeah ? 

Thanks,

**Comment by leblowl** [comment]
@Ahmedouu I think we need to fix this up a bit. I also noticed the backend doesn't seem to initialize everything until it has some interaction from the frontend.

**Comment by Ahmedouu** [comment]
@leblowl 
from what I can tell, this promise doesn't get enough time to resolve when running the backend on it's own:
https://github.com/TryQuiet/quiet/blob/f6b1f19083967fb283a2863348453f92db209fb2/packages/backend/src/backendManager.ts#L53C1-L66C4
I added some logging statement after it and they only show when running the desktop, and I think it's time issue cause I wrapped this in setTimeout and the logs appear when running the backend.

**Comment by leblowl** [comment]
@Ahmedouu Interesting, if you have any ideas for fixes, they are always welcome!

**Comment by Ahmedouu** [comment]
@leblowl  thanks for replying, i am not sure yet but I tried using electron instead of the node binary to run the backend process:
```
npm run webpack && DEBUG=* ../desktop/node_modules/electron/dist/electron ../backend-bundle/bundle.cjs -d 42241 -a '/home/perky/.config/Quietdev' -r '/home/perky/quiet/packages/desktop/node_modules/electron
/dist/resources' -p desktop

//in backendManager.ts 
process.env.ELECTRON_RUN_AS_NODE = '1'; 
```
unfortunately though I kept running into this error https://github.com/electron/electron/issues/40394, it's worth noting though that there is a warning in the desktop run about a sandboxed_renderer.bundle.js and several others sandbox bundles my guess is that it's related to this.

at the moment i am not sure how to debug this but will keep you updated when i get the chance.

**Comment by holmesworcester** [comment]
@EmiM or @vinkabuki -- do either of you know how to do this? 

We have tests that run backend and state-manager together without Electron, correct? Can you point to an example here?

**Comment by vinkabuki** [comment]
> when I run the backend I don't see libp2p or Tor starting, it always stops here: `backend:SocketService Data server running on port 33947 +1ms`
> 
> When running the desktop there is libva error but it doesn't stop it keeps going until tor
> 
> please advise how to go around this or should I just wait for the fix

That's correct, libp2p, ipfs, storage don't start automatically, you must create a community and run launchCommunity with these params, first you must create all of those, check launchCommunity.saga.ts and connections-manager.service.ts
  const payload: InitCommunityPayload = {
    id: identity.id,
    peerId: identity.peerId,
    hiddenService: identity.hiddenService,
    peers: peerList,
  }
